File: SpreadCheetahFSharpSamples/DisposeAsync.fs
module SpreadCheetahFSharpSamples.DisposeAsync

open System.IO
open SpreadCheetah

let sample() = task {
    use stream = File.Create("dispose-async.xlsx")

    // SpreadCheetah also similarly implements DisposeAsync.
    use! spreadsheet = Spreadsheet.CreateNewAsync(stream)

    let row = [|
        Cell("Answer to the ultimate question:")
        Cell(42)
    |]

    do! spreadsheet.StartWorksheetAsync("Sheet 1")
    do! spreadsheet.AddRowAsync(row)
    do! spreadsheet.FinishAsync()
}

File: SpreadCheetahSamples/DisposeAsync.cs
using SpreadCheetah;

namespace SpreadCheetahSamples;

public static class DisposeAsync
{
    public static async Task Sample()
    {
        // From C# 8.0 and later, streams implement DisposeAsync and can be disposed with `await using`.
        await using var stream = File.Create("dispose-async.xlsx");

        // SpreadCheetah also similarly implements DisposeAsync.
        await using var spreadsheet = await Spreadsheet.CreateNewAsync(stream);

        Cell[] row =
        [
            new("Answer to the ultimate question:"),
            new(42)
        ];

        await spreadsheet.StartWorksheetAsync("Sheet 1");
        await spreadsheet.AddRowAsync(row);
        await spreadsheet.FinishAsync();
    }
}
